AutoCollection/HttpRequestParser.ts

```diff
@@ -167,13 +165,15 @@ class HttpRequestParser extends RequestParser {
         }
 
         var encrypted = (<any>request).connection ? ((<any>request).connection as any).encrypted : null;
-        var requestUrl = url.parse(request.url);
+
+        var protocol = (encrypted || request.headers["x-forwarded-proto"] == "https") ? "https" : "http";
```

Comment:
Update the sequence to get the protocol first, then URL API can be used - it requires protocol in url.